sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala

```diff
@@ -199,6 +199,15 @@ class StreamingQueryManager private[sql] (sparkSession: SparkSession) extends Lo
     listenerBus.removeListener(listener)
   }
 
+  /**
+   * Deregister all [[StreamingQueryListener]]s attached to this [[StreamingQueryManager]].
+   *
+   * @since 3.0.0
+   */
+  def removeAllListeners(): Unit = {
+    listenerBus.removeAllListeners()
```

Comment:
Hm, is it safe? Looks it's going to remove all other listeners in Spark Context. So, seems like if there is one Spark Context is shared for multiple shells (like Zeppelin), calling this API will break other shells if they registered their own custom listeners.

Reply:
Hi @HyukjinKwon, thanks for sharing your concerns. I've looked through this and I agree with you; I think I'm trying to use this as a workaround for not properly keeping track of my streaming query listeners. At some point in the future this may still be a useful thing to expose but for now I'll close this PR.

cc @jose-torres

Reply:
If use cases like zepplin want isolation between users, then they should use the isolation mechanism already present (i.e. `SparkSession.newSession`), right?

Reply:
Oh, I misunderstood that it remove all listeners from its Spark context. Yes, I am fine with it in that case.